pkg/login/social/okta_oauth.go

```diff
@@ -0,0 +1,126 @@
+package social
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/grafana/grafana/pkg/models"
+	"golang.org/x/oauth2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+type SocialOkta struct {
+	*SocialBase
+	apiUrl            string
+	allowedGroups     []string
+	roleAttributePath string
+}
+
+type OktaClaims struct {
+	ID                string `json:"sub"`
+	Email             string `json:"email"`
+	PreferredUsername string `json:"preferred_username"`
+	Name              string `json:"name"`
+}
+
+type OktaUserInfoJson struct {
+	Name        string              `json:"name"`
+	DisplayName string              `json:"display_name"`
+	Login       string              `json:"login"`
+	Username    string              `json:"username"`
+	Email       string              `json:"email"`
+	Upn         string              `json:"upn"`
+	Attributes  map[string][]string `json:"attributes"`
+	Groups      []string            `json:"groups"`
+	rawJSON     []byte
+}
+
+func (s *SocialOkta) Type() int {
+	return int(models.OKTA)
+}
+
+func (s *SocialOkta) UserInfo(client *http.Client, token *oauth2.Token) (*BasicUserInfo, error) {
+	idToken := token.Extra("id_token")
+	if idToken == nil {
+		return nil, fmt.Errorf("No id_token found")
+	}
+
+	parsedToken, err := jwt.ParseSigned(idToken.(string))
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing id token")
+	}
+
+	var claims OktaClaims
+	if err := parsedToken.UnsafeClaimsWithoutVerification(&claims); err != nil {
+		return nil, fmt.Errorf("Error getting claims from id token")
+	}
+
+	email := claims.extractEmail()
+
+	if email == "" {
+		return nil, errors.New("Error getting user info: No email found in access token")
```

Comment:
Errors should not be [capitalized](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).